internal/server/proto/server.proto

```diff
@@ -2393,6 +2413,60 @@ message Artifact {
   google.protobuf.Any artifact = 1;
 }
 
+/********************************************************************
+* Ondemand Runners
+********************************************************************/
+message OndemandRunner {
+  // id is the unique ID of the runner config
+  string id = 1;
+
+  // oci_url is the OCI image that will be used to boot the ondemand runner.
+  string oci_url = 2;
+
+  // environment_variables is any env vars that should be exposed to the
+  // ondemand runner. This does not need to include any WAYPOINT specific
+  // variables, those are automatically added.
+  map<string, string> environment_variables = 3;
+
+  // plugin type is used to launch the plugin to start the batch task.
+  string plugin_type = 4;
+
+  // plugin config is the configuration for the plugin that is created. It
+  // is usually HCL and is decoded like the other plugins, and is plugin
+  // specific.
+  bytes plugin_config = 5;
+
+  // config format specifies the format of plugin_config
+  Project.Format config_format = 6;
+
+  // Indicates if this entry is the default for any new projects.
+  bool default = 7;
+}
+
+message UpsertOndemandRunnerRequest {
+  // ondemand_runner to upsert. If the id is empty, then this is an insert,
+  // otherwise this is an update operation.
+  OndemandRunner ondemand_runner = 1;
+}
+
+message UpsertOndemandRunnerResponse {
+  // The resulting ondemand runner value. It should replace the one that was
+  // sent in the request.
+  OndemandRunner ondemand_runner = 1;
+}
+
+message GetOndemandRunnerRequest {
+  Ref.OndemandRunner ondemand_runner = 1;
+}
+
+message GetOndemandRunnerResponse {
+  OndemandRunner ondemand_runner = 1;
```

Comment:
If we rename all these to have `config` then I think this field can just be `config`